pkg/registries/ecr/transport.go

```diff
@@ -0,0 +1,70 @@
+package ecr
+
+import (
+	"encoding/base64"
+	"net/http"
+	"strings"
+	"time"
+
+	awsECR "github.com/aws/aws-sdk-go/service/ecr"
+	"github.com/heroku/docker-registry-client/registry"
+	"github.com/pkg/errors"
+	"github.com/stackrox/rox/pkg/registries/docker"
+	"github.com/stackrox/rox/pkg/sync"
+)
+
+type awsTransport struct {
+	registry.Transport
+	config    *docker.Config
+	client    *awsECR.ECR
+	expiresAt *time.Time
+	mutex     sync.Mutex
+}
+
+func newAWSTransport(config *docker.Config, client *awsECR.ECR) *awsTransport {
+	transport := &awsTransport{config: config, client: client}
+	if err := transport.refreshNoLock(); err != nil {
+		log.Error("Failed to refresh token: ", err)
+	}
+	return transport
+}
+
+func (t *awsTransport) RoundTrip(req *http.Request) (*http.Response, error) {
+	t.mutex.Lock()
+	defer t.mutex.Unlock()
```

Comment:
is it ok to hold the lock for the duration of the `RoudTrip`? I wonder if it'd be better to copy the transport and use the copy's `RoundTrip`, so only the refresh is locked

Reply:
See my comment in the next PR

Reply:
resolving, as there is a followup PR to address this